### DISCUSSION TITLE
Unable to create new project on Windows 11

### DISCUSSION THREAD

**Discussion by tlgould** [thegetty/quire#962] [Get Help]
Hello,

I've been following tutorials to install and build a project using Quire following the v1 install docs at https://quire.getty.edu/docs-v1/install-uninstall/ but coming across problems when I try to create a new project. Info as follows:

**OS:** Windows 11 23H2
**Node:** v18.20.4
**NVM:** 1.1.10
**Git:** version 2.39.0.windows.2

When I try to run _quire new_, I get the following output:

> quire new test
[CLI:quire] init-starter
  project root: "test"
  starter: "https://github.com/thegetty/quire-starter-default"
test set to use quire-11ty@1.0.0-rc.17
[CLI:quire] installing @thegetty/quire-11ty@1.0.0-rc.17 into test
Error encountered while installing @thegetty/quire-11ty@1.0.0-rc.17 NPM package.
Error: ENOENT: no such file or directory, mkdir 'test\node_modules\.install-npm-version-temp-BZw2aC\node_modules\@thegetty\quire-11ty\node_modules'

I can see that the folder structure has been created, but no files have been added in the node_modules folder. When I attempt to build or preview, I get a missing modules error (as it's expecting to find @11ty in the node_modules folder):

> [CLI] running eleventy using lib/11ty cli
[CLI:11ty] running eleventy serve
node:internal/modules/cjs/loader:1143
  throw err;
  ^
Error: Cannot find module 'C:\Users\terenceg\catalog\sample2\node_modules\@11ty\eleventy\cmd.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
    at Module._load (node:internal/modules/cjs/loader:981:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
Node.js v18.20.4

Wondered if any other Windows users had come across this before, and could suggest possible workarounds?

Thank you!

**Comment by Erin-Cecele** [comment]
Hi @tlgould, while inside the `test` folder, can you run `npm install` and see if that helps with the missing node modules?

**Reply by tlgould** [reply]
Hi @Erin-Cecele ,

Thanks for the advice - I'm still a little new to Node/NPM so this is a bit of a learning curve.

When I try to run _npm install_ inside the directory, it also produces an error as follows:

> test2> npm install
npm error code ENOENT
npm error syscall open
npm error path C:\Users\terenceg\quire_projects\test2\package.json
npm error errno -4058
npm error enoent Could not read package.json: Error: ENOENT: no such file or directory, open 'C:\Users\terenceg\quire_projects\test2\package.json'
npm error enoent This is related to npm not being able to find a file.
npm error enoent
npm error A complete log of this run can be found in: C:\Users\terenceg\AppData\Local\npm-cache\_logs\2024-10-15T22_06_38_256Z-debug-0.log`

When I review the folder, I can see that no package.json file was created using _quire new_, which might explain why the error is being thrown?

I've also tried this:
1. create new project test 2, get error.
2. Download a copy of package.json from [thegetty/quire-starter-default](https://github.com/thegetty/quire-starter-default) and add to folder, then run npm install
3. It completes, and can see modules installed, but @11ty is still mising, so get the cannot find module error.

Would it be helpful to share an example of what is created when I run _quire new_?

Thanks,

TG

**Comment by Erin-Cecele** [comment]
Hi @tlgould this looks like the same issue that @metro-glenn had in this issue: https://github.com/thegetty/quire/discussions/724. Looks like he tried running `quire new [project-name]` again and the second time all the files were correctly installed. Can you please try starting a new project again?

**Reply by Erin-Cecele** [reply]
I see you tried running a new project called `test 2` but please note there can be no spaces... so it should be `quire new test-2`. Let me know if that helps!

**Reply by tlgould** [reply]
Hi @Erin-Cecele ,

Unfortunately, these steps dont appear to make any difference. Every time I try to start a new project, I get the same error:

>PS C:\Users\terenceg\quire_projects> quire new test1
[CLI:quire] init-starter
  project root: "test1"
  starter: "https://github.com/thegetty/quire-starter-default"
test1 set to use quire-11ty@1.0.0-rc.17
[CLI:quire] installing @thegetty/quire-11ty@1.0.0-rc.17 into test1
Error encountered while installing @thegetty/quire-11ty@1.0.0-rc.17 NPM package.
Error: ENOENT: no such file or directory, mkdir 'test1\node_modules\.install-npm-version-temp-0czwSd\node_modules\@thegetty\quire-11ty\node_modules'
PS C:\Users\terenceg\quire_projects> quire new test2
[CLI:quire] init-starter
  project root: "test2"
  starter: "https://github.com/thegetty/quire-starter-default"
test2 set to use quire-11ty@1.0.0-rc.17
[CLI:quire] installing @thegetty/quire-11ty@1.0.0-rc.17 into test2
Error encountered while installing @thegetty/quire-11ty@1.0.0-rc.17 NPM package.
Error: ENOENT: no such file or directory, mkdir 'test2\node_modules\.install-npm-version-temp-xDRJh6\node_modules\@thegetty\quire-11ty\node_modules'
PS C:\Users\terenceg\quire_projects>

Note I have also tried running _quire new test1_ again, which results in an error (but I expect this as you cant create a new project in an existing directory?)

>PS C:\Users\terenceg\quire_projects> quire new test1
[CLI] cannot create a starter project in test1 because it is not empty
[CLI:quire] installing @thegetty/quire-11ty@undefined into test1
[CLI:error]  GitError: fatal: pathspec 'package.json' did not match any files
    at Object.action (file:///C:/Users/terenceg/AppData/Roaming/npm/node_modules/@thegetty/quire-cli/node_modules/simple-git/dist/esm/index.js:4462:25)
    at PluginStore.exec (file:///C:/Users/terenceg/AppData/Roaming/npm/node_modules/@thegetty/quire-cli/node_modules/simple-git/dist/esm/index.js:4501:25)
    at file:///C:/Users/terenceg/AppData/Roaming/npm/node_modules/@thegetty/quire-cli/node_modules/simple-git/dist/esm/index.js:1369:43
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (file:///C:/Users/terenceg/AppData/Roaming/npm/node_modules/@thegetty/quire-cli/node_modules/simple-git/dist/esm/index.js:1367:16)
    at GitExecutorChain.<anonymous> (file:///C:/Users/terenceg/AppData/Roaming/npm/node_modules/@thegetty/quire-cli/node_modules/simple-git/dist/esm/index.js:1351:44)
    at Generator.next (<anonymous>)
    at fulfilled (file:///C:/Users/terenceg/AppData/Roaming/npm/node_modules/@thegetty/quire-cli/node_modules/simple-git/dist/esm/index.js:45:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  task: {
    commands: [ 'rm', '-f', 'package.json' ],
    format: 'utf-8',
    parser: [Function: parser]
  }
}
C:\Users\terenceg\AppData\Roaming\npm\node_modules\@thegetty\quire-cli\node_modules\install-npm-version\lib\Npm.js:39
                    logger.Write(`Package name is "${packageMetadata.name}".`, Verbosity_1.Verbosity.Debug);
                                                                     ^
TypeError: Cannot read properties of undefined (reading 'name')
    at C:\Users\terenceg\AppData\Roaming\npm\node_modules\@thegetty\quire-cli\node_modules\install-npm-version\lib\Npm.js:39:70
    at C:\Users\terenceg\AppData\Roaming\npm\node_modules\@thegetty\quire-cli\node_modules\npm\node_modules\call-limit\call-limit.js:28:43
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)

Node.js v18.20.4
PS C:\Users\terenceg\quire_projects>

Please note I have tried alternate names for the top level directory in place of quire_projects and also tried from my home directory to same effect.

Running quire new [project-name] results in the following files being added to the folder:

<img width="576" alt="image" src="https://github.com/user-attachments/assets/4cb85b03-f928-4a2c-94e2-73f77ff55db9">

the node_modules folder only containng a hierarchy of empty folders:

─node_modules
    └───.install-npm-version-temp-nKOBQM
        └───node_modules

**Comment by Erin-Cecele** [comment]
Hi @tlgould I have a workaround that I just did with another Windows user. Can you email me at quire@getty.edu so I can help you get up-and-running with a new project?

**Reply by tlgould** [reply]
Hi @Erin-Cecele ,

Just to say I dropped a line via email, so hopefully that's come through.

**Comment by rtrepagn** [comment]
I am also having this same issue and it would be great if anyone had a solution to this problem.

**Reply by Erin-Cecele** [reply]
Hi @rtrepagn, please email me at quire@getty.edu and I can help get you up and running!

**Reply by Erin-Cecele** [reply]
Apologies, Renee. I see you have already emailed me! I didn't put the two user names together. I just responded. Thank you!

**Comment by cbutcosk** [comment]
@Erin-Cecele This issue is being caused by a security patch in node ( https://nodejs.org/en/blog/vulnerability/april-2024-security-releases-2 ) for which an upstream dependency (`install-npm-version`) has not yet been updated to account. On windows `quire new` uses the dependency to spawn a process that installs a project's basic 11ty code, but currently the spawned process silently errors, leading to the symptoms (mostly-empty project directories) you and others in this thread are experiencing.

Now that it's identified it should be fixed in a forthcoming release. Unfortunately the quick fix outlined in that article (using `--security-revert=CVE-2024-27980`) does not work in this instance for some reason, possibly because a library that quire uses spawns the process and not quire-cli directly.

**Comment by rtrepagn** [comment]
Dear Erin and Charlie,
What should I do in the meantime? Do I re-download and update node?

Best,
Renee Trepagnier

**Comment by Erin-Cecele** [comment]
Hi Renee, we are working on a fix for this now that will hopefully be included in an upcoming release. We will keep you updated. If you are looking to get started ASAP, you can use the sample project that I sent you via WeTransfer.

**Comment by cbutcosk** [comment]
@Erin-Cecele @rtrepagn 
The release candidate that fixes this issue is in QA and testing on npm as version `1.0.0-rc.16` of `@thegetty/quire-cli`. Folks in this thread might give it a try -- use `npm install -g @thegetty/quire-cli@^1.0.0-rc.16` to install it globally and `quire new test` should succeed.[^1]

[^1]: ~Users installing the CLI local to a publication / project (much less common) will need to add `--install-stragey=nested` to the npm install command or their .npmrc file in addition to removing `-g` for the patch to be applied correctly.~ -- erm, discovered a bug in project-local CLI installs actually

**Reply by Erin-Cecele** [reply]
@cbutcosk I was in the process of updating the Quire documentation to include the instructions for installing locally. Since it looks like you discovered a bug, I will hold off. Please let me know when we have the correct command available for documentation.

**Comment by tlgould** [comment]
Hello @cbutcosk ,

 I've given RC16 a quick try on a Win11 laptop  just now and I have been able to successfully create a new project and run a preview without any errors (some deprecation warnings, but the site does render). I'd be very happy to give it more of a try in next couple of days and report back if that would be helpful?

**Reply by Erin-Cecele** [reply]
That is great news @tlgould!! We would gladly welcome additional testing and feedback. Thank you!
